### PR TITLE
Fix iterator crash and leftover edges

### PR DESCRIPTION
**Pull request by joeyballentine**

### CODE REVIEW

File: src/renderer/components/inputs/InputContainer.tsx [hunk 1]
```diff
@@ -108,12 +108,28 @@ export const InputContainer = memo(
...
-                    .outputDefaults.get(parentOutputId)!;
-                return getTypeAccentColors(parentType, typeDefinitions, isDarkMode)[0];
+                if (parentNode) {
+                    const parentDef = functionDefinitions.get(parentNode.data.schemaId);
+                    if (!parentDef) {
+                        return getTypeAccentColors(
+                            NeverType.instance,
+                            typeDefinitions,
+                            isDarkMode
+                        )[0];
```
Comment (RunDevelopment) [inline src/renderer/components/inputs/InputContainer.tsx:120]: This seems like a lot to get the default color. Maybe we could just export the default color and use it directly?

File: src/renderer/contexts/GlobalNodeState.tsx [hunk 2]
```diff
@@ -599,6 +599,21 @@ export const GlobalProvider = memo(
             [changeNodes, changeEdges]
         );
 
+        // TODO: This shouldn't be necessary, but it at least fixes the issue
```
Comment (RunDevelopment) [inline src/renderer/contexts/GlobalNodeState.tsx:602]: So why is it necessary?

Comment (joeyballentine) [reply src/renderer/contexts/GlobalNodeState.tsx:602]: Deleting an iterator doesn't delete an edge if the edge is connected to an outer node. I couldn't think of a way we could do this in removeNodeById, so I just did it here

Comment (RunDevelopment) [reply src/renderer/contexts/GlobalNodeState.tsx:602]: Ah, I forget to account for iterator helper nodes in the code snippet, but the general idea should be clear.

File: src/renderer/components/inputs/InputContainer.tsx [hunk 3]
```diff
@@ -108,12 +108,20 @@ export const InputContainer = memo(
...
+                    if (!parentDef) {
+                        return defaultColor;
+                    }
+                    const parentType = parentDef.outputDefaults.get(parentOutputId);
+                    if (!parentType) {
+                        return defaultColor;
+                    }
+                    return getTypeAccentColors(parentType, typeDefinitions, isDarkMode)[0];
+                }
+                return getTypeAccentColors(NeverType.instance, typeDefinitions, isDarkMode)[0];
```
Comment (RunDevelopment) [inline src/renderer/components/inputs/InputContainer.tsx:124]: This should have been another `return defaultColor;` but whatever.

Comment (joeyballentine) [reply src/renderer/components/inputs/InputContainer.tsx:124]: Ah shit. Didn't notice it because of the formatting

Comment (joeyballentine) [reply src/renderer/components/inputs/InputContainer.tsx:124]: I knew I should have left this open for another check but I was confident I had fixed everything... Oh well

**Pull request merged**